### PR TITLE
Update coding style with test method annotation

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Update coding style with test method annotation, e.g. if we want to add a test method say 'foo' to an object, we will name it with prefix TEST, TEST_foo.

### CODE REVIEW

File: CODING_STYLE.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ line width, indentation and ordering (for includes, using directives and etc). 
 * Use **snake_case** for namespace names and build targets
 * Use **UPPER_SNAKE_CASE** for macros.
 * Use **kPascalCase** for static constants and enumerators.
-
+* Use **TEST_** prefix for test only class methods, e.g. obj.TEST_foo().
```
Comment (majetideepak) [inline CODING_STYLE.md:29]: This makes sense, but do we have examples of test-only class methods?

Comment (aditi-pandit) [reply CODING_STYLE.md:29]: Yeah, I'm not sure what test only class methods means. Are these methods that are added to classes for exclusive use in tests ? Would something like PlanBuilder or QueryAssertions be considered test only ?

**Comment by aditi-pandit** [comment]
Is this PR still being worked on ?

File: CODING_STYLE.md [hunk 2]
```diff
@@ -57,7 +57,12 @@ line width, indentation and ordering (for includes, using directives and etc). 
 * Use **snake_case** for namespace names and build targets
 * Use **UPPER_SNAKE_CASE** for macros.
 * Use **kPascalCase** for static constants and enumerators.
-
+* Use **testing** prefix for test only class methods, e.g. obj.testingFoo().
+  User should try to test the object using its public APIs if possible, and
```
Comment (pedroerp) [inline CODING_STYLE.md:61]: Perhaps a small rephrasing:

> As much as possible, refrain from adding test methods, and test objects using their public APIs. Only use test methods in rare edge cases where this is not possible. 

> For example, `MemoryAllocator::testingSetFailureInjection()` is used to to inject various memory allocation failures to test error handling paths.

**Review by pedroerp — APPROVED** [review]

File: CODING_STYLE.md (hunk 2 above)
Comment (xiaoxmeng) [reply CODING_STYLE.md:61]: Thanks!

Comment (mbasmanova) [reply CODING_STYLE.md:61]: +1

**Review by mbasmanova — APPROVED** [review]
@xiaoxmeng Looks good to me. Let's make sure this is used sparingly.

**Pull request closed without merging**